ipa/prover.go

```diff
@@ -17,6 +17,17 @@ func init() {
 	maxEvalPointInsideDomain.SetUint64(common.VectorLength - 1)
 }
 
+var (
+	labelDomainSep   = []byte("ipa")
+	labelC           = []byte("C")
+	labelInputPoint  = []byte("input point")
+	labelOutputPoint = []byte("output point")
+	labelW           = []byte("w")
+	labelL           = []byte("L")
+	labelR           = []byte("R")
+	labelX           = []byte("x")
```

Comment:
To avoid doing these expressions in every call site, we define them as unexported globals. If we don't do this, every time we append, e.g, a `[]byte("input point")` to the transcript, that will mean an allocation.

Considering the labels are never mutated by the transcript (i.e: only appended to a sha256-state), this is fine (i.e: they can be reused).